src/PowerShellEditorServices.Protocol/Server/DebugAdapter.cs

```diff
@@ -678,11 +678,23 @@ protected override void Shutdown()
             }
             else
             {
-                VariableDetails result =
-                    await editorSession.DebugService.EvaluateExpression(
+                VariableDetailsBase result;
+                if (evaluateParams.Context.Equals("watch", StringComparison.OrdinalIgnoreCase))
+                {
+                    // If we evaluate an expression from a watch context then the server goes into a deadlock.
+                    // So instead, we retrieve the value from already evaluated variables.
+                    result = editorSession.DebugService.GetVariableFromExpression(
```

Comment:
Keep in mind that the user can enter arbitrary expressions - not just variable names e.g. `1+1`, `Get-Date`, `$count+10`